pkg/config/configobserver/configobserver.go

```diff
@@ -77,9 +78,15 @@ func (c *Controller) retrieveToken() error {
 			if err := json.Unmarshal(data, &pullSecret); err != nil {
 				klog.Errorf("Unable to unmarshal cluster pull-secret: %v", err)
 			}
-			if auth, ok := pullSecret.Auths["cloud.openshift.com"]; ok && len(auth.Auth) > 0 {
-				klog.V(4).Info("Found cloud.openshift.com token")
-				nextConfig.Token = auth.Auth
+			if auth, ok := pullSecret.Auths["cloud.openshift.com"]; ok {
+				token := strings.TrimSpace(auth.Auth)
+				if strings.Contains(token, "\n") || strings.Contains(token, "\r") {
+					return fmt.Errorf("cluster authorization token is not valid: contains newlines")
+				}
+				if len(token) > 0 {
```

Comment:
how about empty token?

Reply:
I don't know how many users out there have edited their pull secrets to remove the token but didn't remove the key.  I don't feel strongly, can make this be an error, but since only telemetry and insights use it I was unsure how many we would have in prod that behaved this way.  Also, in docker serialization empty and absent are the same (in most of the places that consume this) so it's likely there are people who would remove the value but not the key.